pyo3-introspection/src/introspection.rs

```diff
@@ -65,18 +70,18 @@ fn parse_chunks(chunks: &[Chunk], main_module_name: &str) -> Result<Module> {
 }
 
 fn convert_module(
+    id: &str,
     name: &str,
     members: &[String],
-    consts: &[ConstChunk],
     incomplete: bool,
     chunks_by_id: &HashMap<&str, &Chunk>,
     chunks_by_parent: &HashMap<&str, Vec<&Chunk>>,
 ) -> Result<Module> {
-    let (modules, classes, functions) = convert_members(
-        &members
+    let (modules, classes, functions, attributes) = convert_members(
+        members
             .iter()
             .filter_map(|id| chunks_by_id.get(id.as_str()).copied())
-            .collect::<Vec<_>>(),
+            .chain(chunks_by_parent.get(&id).into_iter().flatten().copied()),
```

Comment:
Is it ever the case that `members` and `chunks_by_parent` contain the same thing repeated? Can we merge these two concepts somehow?

Reply:
In the current generated code no, we carefully make sure this is not the case.

The two concepts exists because some elements know their subelements (e.g. modules) whereas some other elements knows their parent elements (e.g. methods) but the inverse is not true (e.g. a class does not know its module). It's indeed a bit ugly.